docs/tasks/administer-cluster/running-cloud-controller.md

```diff
@@ -1,34 +1,85 @@
+* cloud authentication/authorization: your cloud may require a token or IAM rules to allow access to their APIs
+* kubernetes authentication/authorization: cloud-controller-manager may need abac/rbac set to speak to the kubernetes apiserver
+* high availabilty: like kube-controller-manager, you may want a high available setup for cloud controller manager using leader election.
 
 ### Running cloud-controller-manager
 
-To run `cloud-controller-manager`, add it to your existing Kubernetes cluster as a Master component. All other master components except `kube-controller-manager` can be run without any changes.
+Successfully running cloud-controller-manager requires some changes to your cluster configuration.
+
+* `kube-apiserver` and `kube-controller-manager` must either specify no cloud provider or the `external` cloud provider for the flag `--cloud-provider`. This ensures that it does not run any cloud specific loops that would be run by cloud controller manager.
+* `kubelet` must run with `--cloud-provider=external`. This is to ensure that the kubelet is aware that it must be initialized by the cloud controller manager before it is scheduled any work.
+
+Keep in mind that setting up your cluster to use cloud controller manager will change your cluster behaviour in a few ways:
+
+* kubelets specifying `--cloud-provider=external` will add a [taint during initialization](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/scheduler/algorithm/well_known_labels.go#L51-L55). This marks the node as needing a second initialization from an external controller before it can be scheduled work. Note that in the event that cloud controller manager is not available, new nodes in the cluster will be left unscheduable. The taint is important since the scheduler may require cloud specific information about nodes such as it's region or type (high cpu, gpu, high memory, spot instance, etc).
+* cloud information about nodes in the cluster will no longer be retrieved using local metadata, but instead all API calls to retreive node information will go through cloud controller manager. This may mean you can restrict access to your cloud API on the kubelets for better security. For larger clusters you may want to consider if cloud controller manager will hit rate limits since it is now responsible for almost all API calls to your cloud from within the cluster.
+
+
+As of v1.8, cloud controller manager can implement:
+
+* node controller - responsible for updating nodes using cloud APIs.
+* service controller - responsible for loadbalancers on your cloud against services of type LoadBalancer.
+* route controller - responsible for setting up network routes on your cloud
+* persistent volume label controller - responsible for labeling persistent volumes on your cloud - ensure that the persistent volume label admission plugin is not enabled on your kube-apiserver.
+
+
+## Examples
```

Comment:
Can we make a generic variant of this somehow?
```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: cloud-controller-manager
  namespace: kube-system
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1beta1
metadata:
  name: system:cloud-controller-manager
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- kind: ServiceAccount
  name: cloud-controller-manager
  namespace: kube-system
---
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  labels:
    app: cloud-controller-manager
  name: cloud-controller-manager
  namespace: kube-system
spec:
  selector:
    matchLabels:
      app: cloud-controller-manager
  template:
    metadata:
      labels:
        app: cloud-controller-manager
    spec:
      serviceAccountName: cloud-controller-manager
      containers:
      - name: cloud-controller-manager
        image: gcr.io/google_containers/cloud-controller-manager:v1.7.0
        command:
        - /usr/local/bin/cloud-controller-manager
        - --cloud-provider=gce
        - --allocate-node-cidrs=true
        - --configure-cloud-routes=true
        - --cluster-cidr=10.135.0.0/16
        volumeMounts:
        - name: etcssl
          mountPath: /etc/ssl
      tolerations:
      - key: node.cloudprovider.kubernetes.io/uninitialized
        value: "true"
        effect: NoSchedule
      - key: node-role.kubernetes.io/master
        effect: NoSchedule
      volumes:
      - name: etcssl
        hostPath:
          path: /etc/ssl
```

In order to provide an example. For instance, parameterize out `--cloud-provider` etc

Reply:
Let's add an example file somewhere in kubernetes/kubernetes and just link to it

Reply:
@luxas added examples here https://github.com/kubernetes/kubernetes/pull/52374 and I'll link them once they're in master

Reply:
What's the best place to provide comments on this example? I'll put it here but can move it if some where else is better. 

* Is `cluster-admin` still the preferred way to handle the controller manager (and in this case the ccm) as well or should we be using a service account per controller? cc @liggitt 
* The `app` label should be `k8s-app`.
* Why is the example a DS instead of a deployment like the self-hosted controller manager? We can use anti-affinity to make sure they are spread. 
* The `etcssl` volume should be read only.

Reply:
* I'd expect the cloud controller manager to have a defined role, not use cluster-admin
* The CCM also supports (and should use) `--use-service-account-credentials` to run the individual control loops with specific service accounts